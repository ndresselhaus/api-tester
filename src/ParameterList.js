import React from 'react';
import PropTypes from 'prop-types';

const ParameterList = ({title, params = {}, onChange}) => {

    const inputs = Object.keys(params).map(key => {
        const value = params[key];
        return <div key={key}>
            <label>{key}
                <input type="text" value={value} onChange={e => onChange(key, e.target.value)} />
            </label>
        </div>;
    });

    if (!inputs.length) return null;

    return <div>
        <h3>{title}</h3>
        {inputs}
    </div>;
};

ParameterList.propTypes = {
    title: PropTypes.string.isRequired,
    params: PropTypes.object,
    onChange: PropTypes.func.isRequired,
};

export default ParameterList;