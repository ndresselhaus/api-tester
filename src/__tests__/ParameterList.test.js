import React from 'react';
import ReactDOM from 'react-dom';
import ParameterList from '../ParameterList';
import {shallow} from 'enzyme';

const title = 'Test Title';
const params = {a: 1, b: '2'};

it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParameterList title={title} onChange={() => {}} />, div);
});

it('renders params and title for snapshot', () => {
    const wrapper = shallow(<ParameterList title={title} params={params} onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
});

it('calls onChange for text changes', () => {
    const stub = jest.fn();
    const wrapper = shallow(<ParameterList title={title} params={params} onChange={stub} />);

    const event = {target: {value: 'new value'}};
    wrapper.find('input[type="text"]').first().simulate('change', event);
    expect(stub).toBeCalledWith('a', 'new value');
});