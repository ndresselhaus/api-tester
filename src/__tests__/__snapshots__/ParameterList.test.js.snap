// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders params and title for snapshot 1`] = `
<div>
  <h3>
    Test Title
  </h3>
  <div>
    <label>
      a
      <input
        onChange={[Function]}
        type="text"
        value={1}
      />
    </label>
  </div>
  <div>
    <label>
      b
      <input
        onChange={[Function]}
        type="text"
        value="2"
      />
    </label>
  </div>
</div>
`;
